src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * MMDBot - https://github.com/MinecraftModDevelopment/MMDBot
+ * Copyright (C) 2016-2022 <MMD - MinecraftModDevelopment>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ */
+package com.mcmoddev.mmdbot.modules.logging.misc;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.mcmoddev.mmdbot.MMDBot;
+import com.mcmoddev.mmdbot.utilities.Utils;
+import com.mcmoddev.mmdbot.utilities.console.MMDMarkers;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageUpdateEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.utils.MarkdownUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.Color;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.StreamSupport;
+
+/**
+ * Scam detection system
+ * @author matyrobbrt
+ */
+public class ScamDetector extends ListenerAdapter {
+
+    public static final Gson GSON = new GsonBuilder().disableHtmlEscaping().create();
+    public static final String SCAM_LINKS_DATA_URL = "https://phish.sinking.yachts/v2/all";
+
+    public static final Set<String> SCAM_LINKS = Collections.synchronizedSet(new HashSet<>());
+
+    static {
+        new Thread(ScamDetector::setupScamLinks, "Scam link collector").start();
```

Comment:
Could this be hooked up to a `TimerTask` and run in `TaskScheduler`, so that it doesn't have to be manually refreshed _every_ time?

Reply:
Yea I could do that, but it seems useless since that site doesn't update regularly / frequently. I'm pretty sure the last time I saw it being changed was 2 months ago.

Reply:
We can always have it set to run once a week or maybe byweekly if they really don't update that often, I'd do it this way as I have a tendency to forget to do things like this and I'm the bot owner currently at least :)